src/Tests/InstrEngineTests/InstrEngineTests/TestAppAssembler.cs

```diff
@@ -46,5 +43,46 @@ internal static bool AssembleFile(IEmbeddedResourceFile embeddedResource, string
             Assert.IsTrue(p.ExitCode == 0, FormattableString.Invariant($"Assembly failed for '{embeddedResource.FilePath}' : {stdOut} \n\n {stdErr}"));
             return p.ExitCode == 0;
         }
+
+        internal static string AssembleFiles(List<IEmbeddedResourceFile> embeddedResources, string directoryPath, string assemblyName, bool isDebug, bool is64Bit, bool isExe = false)
+        {
+            ProcessStartInfo psi = new ProcessStartInfo();
+            IEmbeddedResourceFile ilasm = EmbeddedResourceUtils.GetTestResourceFile("ilasm.exe");
+            Assert.IsFalse(ilasm == null, "Could not find ilasm.exe");
+            if (ilasm == null)
+            {
+                return null;
+            }
+
+            psi.FileName = ilasm.FilePath;
+            psi.RedirectStandardError = true;
+            psi.RedirectStandardOutput = true;
+            psi.UseShellExecute = false;
+            string debugSwitch = isDebug ? "/DEBUG=OPT" : string.Empty;
+            string bitSwitch = is64Bit ? "/X64" : "/32BITPREFERRED";
+            if (TestUtils.IsArmProcess)
+            {
+                bitSwitch = is64Bit ? "/ARM64" : "/ARM";
+            }
+            string outputType = isExe ? "/EXE" : "/DLL";
+            string extension = isExe ? "exe" : "dll";
+            string outputFile = Path.Combine(directoryPath, FormattableString.Invariant($"{assemblyName}.{extension}"));
+
+            string[] embeddedResourceFilePaths = embeddedResources.Select(x => x.FilePath).ToArray();
+            string ilFileNames = string.Join("\" \"", embeddedResourceFilePaths);
+
+            psi.Arguments = FormattableString.Invariant($"{debugSwitch} {bitSwitch} {outputType} /OUTPUT=\"{outputFile}\" \"{ilFileNames}\"");
+            Process p = Process.Start(psi);
+            p.WaitForExit(5000);
```

Comment:
>5000 [](http://example.com/codeflow?start=26&length=4)

Do we need this timeout?

Reply:
It only happened during debugging. I think it is probably an issue with .NET 6/7 debugging and the Process class. The CLR has special code that runs in the process to handle certain debugging scenarios, and I think they may have introduced a hanging issue here, probably somewhere in the event source code base. That kind of thing has happened before, where the event source takes a lock and gets hung because it is waiting for the debugger to continue. We didn't have the time to look into it, though.